embedded-graphics/src/primitives/sector.rs

```diff
@@ -0,0 +1,798 @@
+    /// The result is not equivalent to `self.center() * 2` because of rounding.
+    fn center_2x(&self) -> Point {
+        // The radius scaled up by a factor of 2 is equal to the diameter
+        let radius = self.diameter.saturating_sub(1);
+
+        self.top_left * 2 + Size::new(radius, radius)
+    }
+
+    /// Return the end angle of the sector
+    fn angle_end(&self) -> Angle {
+        self.angle_start + self.angle_sweep
+    }
+
+    /// Return a `Line` between the sector center and a point on the circumference following a given angle
+    fn line_from_angle(&self, angle: Angle) -> Line {
+        let center = self.center();
+        let radius = Real::from(self.diameter.saturating_sub(1)) / 2.into();
+
+        let point = Point::new(
+            center.x + i32::from(angle.cos() * radius),
+            center.y - i32::from(angle.sin() * radius),
+        );
+
+        Line::new(center, point)
+    }
+
+    pub(crate) fn expand(&self, offset: u32) -> Self {
+        let diameter = self.diameter.saturating_add(2 * offset);
+
+        Self::with_center(self.center(), diameter, self.angle_start, self.angle_sweep)
+    }
+
+    pub(crate) fn shrink(&self, offset: u32) -> Self {
+        let diameter = self.diameter.saturating_sub(2 * offset);
+
+        Self::with_center(self.center(), diameter, self.angle_start, self.angle_sweep)
+    }
+}
+
+impl Primitive for Sector {
+    type PointsIter = Points;
+
+    fn points(&self) -> Self::PointsIter {
+        Points::new(self)
+    }
+}
+
+impl ContainsPoint for Sector {
+    fn contains(&self, point: Point) -> bool {
+        let delta = self.center_2x() - point * 2;
+        let distance = delta.x.pow(2) as u32 + delta.y.pow(2) as u32;
+
+        let threshold = circle::diameter_to_threshold(self.diameter);
+
+        if distance >= threshold {
+            return false;
+        }
+
+        let angle_start = self.angle_start.normalize_from(-ANGLE_90DEG);
+        let angle_end = self.angle_end().normalize_from(-ANGLE_90DEG);
+        let negative_sweep = self.angle_sweep < Angle::zero();
+
+        let line_a = LinearEquation::from_point_angle(self.center(), self.angle_start);
+        let line_b = LinearEquation::from_point_angle(self.center(), self.angle_end());
+        let draw_above_a = (angle_start < ANGLE_90DEG) ^ negative_sweep;
+        let draw_above_b = (angle_end >= ANGLE_90DEG) ^ negative_sweep;
+        let sweep = self.angle_sweep.abs();
+
+        let side_a = line_a.side(&point);
+        let side_b = line_b.side(&point);
+
+        let correct_a_side = draw_above_a ^ (side_a == LineSide::BELLOW);
+        let correct_b_side = draw_above_b ^ (side_b == LineSide::BELLOW);
+
+        if sweep < ANGLE_180DEG {
+            correct_a_side && correct_b_side
+        } else if sweep < ANGLE_360DEG {
+            correct_a_side || correct_b_side
+        } else {
+            true
+        }
+    }
```

Comment:
This is the same code used in `PlaneSectorIterator::next`. I would suggest to add this to `PlaneSector`:
```rust
fn contains(&self, point: Point) -> bool {
    let side_a = self.line_a.side(point);
    let side_b = self.line_b.side(point);

    let correct_a_side = self.draw_above_a ^ (side_a == LineSide::BELLOW);
    let correct_b_side = self.draw_above_b ^ (side_b == LineSide::BELLOW);

    if self.sweep < ANGLE_180DEG {
        correct_a_side && correct_b_side
    } else if self.sweep < ANGLE_360DEG {
        correct_a_side || correct_b_side
    } else {
        true
    }
}
```
And use it in `PlaneSectorIterator::next` and `ContainsPoint::contains`.